### PR TITLE
Cleanup CaptureData

### PR DESCRIPTION
**Pull request by dimitry-**
Since CaptureData is not movable remove all unique_ptrs
which were there for this reason. Also remove some unused
functions and fields.

Test: builds

### CODE REVIEW

File: src/ClientData/include/ClientData/CaptureData.h [hunk 1]
```diff
@@ -250,25 +244,24 @@ class CaptureData {
...
 
   absl::flat_hash_map<uint64_t, orbit_client_protos::FunctionStats> functions_stats_;
 
   absl::flat_hash_map<int32_t, std::string> thread_names_;
 
+  // For each thread, assume sorted by timestamp and not overlapping.
   absl::flat_hash_map<int32_t, std::vector<orbit_client_protos::ThreadStateSliceInfo>>
-      thread_state_slices_;  // For each thread, assume sorted by timestamp and not overlapping.
-  mutable std::unique_ptr<absl::Mutex> thread_state_slices_mutex_ = std::make_unique<absl::Mutex>();
+      thread_state_slices_ GUARDED_BY(thread_state_slices_mutex_);
```
Comment (florian-kuebler) [inline src/ClientData/include/ClientData/CaptureData.h:263]: Thanks for adding this.

**Review by florian-kuebler — APPROVED** [review]
LGTM

File: src/ClientData/include/ClientData/CaptureData.h (hunk 1 above)
Comment (ronaldfw) [reply src/ClientData/include/ClientData/CaptureData.h:263]: +1

**Review by ronaldfw — APPROVED** [review]
LGTM, thanks, nice cleanup!

**Pull request merged**